dask/dataframe/shuffle.py

```diff
@@ -117,11 +117,10 @@ def sort_values(
         raise ValueError("na_position must be either 'first' or 'last'")
     if not isinstance(by, list):
         by = [by]
-    if len(by) > 1 and df.npartitions > 1 or any(not isinstance(b, str) for b in by):
+    if any(not isinstance(b, str) for b in by):
```

Comment:
Looking at the updated diff, it looks like we've already had support for multi-column sorting through the custom sort function logic added in https://github.com/dask/dask/pull/8345; this just removes the check that was blocking us from specifying multiple `by` columns with a multi-partition frame.